ops/model.py

```diff
@@ -1937,8 +1940,22 @@ def from_name(cls, name: str, message: str):
             return cls._statuses[typing.cast(StatusName, name)](message)
 
     @classmethod
-    def register(cls, child: Type['StatusBase']):
-        """Register a Status for the child's name."""
+    def register(cls, child: Type[_StatusClass]) -> Type[_StatusClass]:
+        """Class decorator to register a subclass for lookup using :meth:`from_name`.
+
+        Note: this method is intended for internal use only.
+        It is used to make the six valid Juju statuses available by name.
+
+        .. code-block:: python
+
+            _StatusClass = typing.TypeVar('_StatusClass', bound='StatusBase')
+
+        Args:
+            child: A subclass of StatusBase, with a ``name`` attribute of type ``str``.
+
+        Returns:
+            The decorated class, unmodified.
+        """
```

Comment:
Should the name of the method get an underscore, if it's for internal use only?

cc @benhoyt for the balance of very hypothetically breaking a public-ish method on a public class.

Reply:
I think it would be nice if it had one, and it doesn't seem to be used elsewhere, but perhaps that's out of scope for this PR?

Reply:
Ah, I see this has been discussed already.